src/t/d17.clj

```diff
@@ -0,0 +1,258 @@
+(ns t.d17
+  (:gen-class)
+  (:require
+    [clojure.math :as math]
+    [clojure.string :as cljstr]
+    [clojure.set :as cljset]
+    [clojure.core.match :refer [match]]))
+
+(def sample
+"Register A: 729
+Register B: 0
+Register C: 0
+
+Program: 0,1,5,4,3,0")
+(def sol "4,6,3,5,6,3,5,2,1,0")
+
+(def sample2
+"Register A: 2024
+Register B: 0
+Register C: 0
+
+Program: 0,3,5,4,3,0")
+(def sol2 117440)
+
+(defn parse-input
+  [input]
+  (let [ [_ A B C instr] (re-find #"Register A: (\d+)\nRegister B: (\d+)\nRegister C: (\d+)\n\nProgram: ([\d,?]+)" input)
+        instr (->> (cljstr/split instr #",")
+                   (map parse-long)
+                   vec
+                   )]
+    [(parse-long A) (parse-long B) (parse-long C) instr]))
+
+
+(defn combo-oprand
+  [A B C oprand]
+  (cond
+    (<= oprand 3) oprand
+    (= 4 oprand) A
+    (= 5 oprand) B
+    (= 6 oprand) C
+    "7 or none of the above is pbm" (prn "Problem with oprand" oprand)))
+
+(defn adv
+  [A B C oprand p]
+  [(quot A (int (math/pow 2 (combo-oprand A B C oprand))))
+   B
+   C
+   nil
+   (+ 2 p)])
+
+(defn bxl
+  [A B C oprand p]
+  [A
+   (bit-xor B oprand)
+   C
+   nil
+   (+ 2 p)])
+
+(defn bst
+  [A B C oprand p]
+  [A
+   (mod (combo-oprand A B C oprand) 8)
+   C
+   nil
+   (+ 2 p)])
+
+(defn jnz
+  [A B C oprand p]
+  [A
+   B
+   C
+   nil
+   (if (zero? A) (+ 2 p) oprand)])
+
+(defn bxc
+  [A B C _ p]
+  [A
+   (bit-xor B C)
+   C
+   nil
+   (+ 2 p)])
+
+(defn out
+  [A B C oprand p]
+  [A
+   B
+   C
+   (mod (combo-oprand A B C oprand) 8)
+   (+ 2 p)])
+
+(defn bdv
+  [A B C oprand p]
+  [A
+   (quot A (int (math/pow 2 (combo-oprand A B C oprand))))
+   C
+   nil
+   (+ 2 p)])
+
+(defn cdv
+  [A B C oprand p]
+  [A
+   B
+   (quot A (int (math/pow 2 (combo-oprand A B C oprand))))
+   nil
+   (+ 2 p)])
+
+(def opcode-to-fn
+  {0 adv
+   1 bxl
+   2 bst
+   3 jnz
+   4 bxc
+   5 out
+   6 bdv
+   7 cdv })
```

Comment:
If all your keys are going to be consecutive integers that start at zero, perhaps a vector works as well?
```suggestion
(def opcode-to-fn [adv bxl bst jnz bxc out bdv cdv])
```

Reply:
true, I think it helped me build it step by step to also see the numbers next to the function (make sure I don't miss one)